server/controllers/finance/cashboxes/currencies.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * @overview Cashboxes/Currencies
+ *
+ * @description
+ * Provides an interface for interacting with currencied accounts attached to
+ * the cashboxes.
+ *
+ * @requires db
+ * @requires NotFound
+ * @requires Topic
+ */
+
+'use strict';
+
+const db = require('../../../lib/db');
+const NotFound = require('../../../lib/errors/NotFound');
+const Topic = require('../../../lib/topic');
+
+exports.list = list;
+exports.detail = detail;
+exports.create = create;
+exports.update = update;
+
+/**
+ * @method list
+ *
+ * @description
+ * Lists the currencied accounts associated with a given cashbox.
+ *
+ * GET /cashboxes/:id/currencies
+ */
+function list(req, res, next) {
+  let sql =
+    `SELECT id, currency_id, account_id, transfer_account_id
+    FROM cash_box_account_currency WHERE cash_box_id = ?;`;
+
+  db.exec(sql, [req.params.id])
+  .then(function (rows) {
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+}
+
+/**
+ * @method detail
+ *
+ * @description
+ * Get the details of a single currencied account associated with the cashbox.
+ *
+ * GET /cashboxes/:id/currencies/:currencyId
+ */
+function detail(req, res, next) {
+  let sql =
+    `SELECT id, account_id, transfer_account_id
+    FROM cash_box_account_currency
+    WHERE cash_box_id = ? AND currency_id = ?;`;
+
+  db.exec(sql, [req.params.id, req.params.currencyId])
+  .then(function (rows) {
+    if (!rows.length) {
```

Comment:
We have a lot of different ways of doing this on the server, some examples; 
```js
if (rows.length === 0) { //...
if (!rows.length) { //...

// Contrived 
let LENGTH_EMPTY = 0;
if (rows.length === LENGTH_EMPTY) { //...

let isEmpty = rows.length === 0;
if (isEmpty) { //...
```

I suggest that we tackle trying to make this uniform now before server business logic becomes more complex going forward.

Something like
```js 
if (utility.isEmpty(databaseQueryResult)) { //...
```

What do you think?
@IMA-WorldHealth/local-contributors